src/nvim/ex_docmd.c

```diff
@@ -1335,25 +1419,108 @@ static char_u * do_one_cmd(char_u **cmdlinep,
    * is equal to the lower.
    */
 
+  if (!IS_USER_CMDIDX(ea.cmdidx)) {
+    if (ea.cmdidx != CMD_SIZE && ea.cmdidx >= 0) {
+      ea.addr_type = cmdnames[(int)ea.cmdidx].cmd_addr_type;
+    } else {
+      ea.addr_type = ADDR_LINES;
+      // ea.addr_type for user commands is set by find_ucmd
+    }
+  // :wincmd range depends on the argument
+  } else if (ea.cmdidx == CMD_wincmd) {
```

Comment:
Was it because of the ea.cmdidx >= 0 check (which I just removed), or something else? I don't see it.